examples/knative/powershell/kn-ps-slack-vsphere-alarm/handler.ps1

```diff
@@ -0,0 +1,146 @@
+Function Process-Init {
+   Write-Host "$(Get-Date) - Processing Init`n"
+
+   Write-Host "$(Get-Date) - Init Processing Completed`n"
+}
+
+Function Process-Shutdown {
+   Write-Host "$(Get-Date) - Processing Shutdown`n"
+
+   Write-Host "$(Get-Date) - Shutdown Processing Completed`n"
+}
+
+Function Process-Handler {
+   param(
+      [Parameter(Position=0,Mandatory=$true)][CloudNative.CloudEvents.CloudEvent]$CloudEvent
+   )
+
+   # Decode CloudEvent
+   try {
+      $cloudEventData = $cloudEvent | Read-CloudEventJsonData -Depth 10
+   } catch {
+      throw "`nPayload must be JSON encoded"
+   }
+
+   if(${env:FUNCTION_DEBUG} -eq "true") {
+      Write-Host "$(Get-Date) - DEBUG: K8s Secrets:`n${env:SLACK_SECRET}`n"
+
+      Write-Host "$(Get-Date) - DEBUG: CloudEventData`n $(${cloudEventData} | Out-String)`n"
+   }
+
+   # Check the CloudEvent envelope properties to make sure this is vSphere Alarm Event AND it is our enriched vSphere Alarm
+   if($CloudEvent.type -eq "com.vmware.event.router/event.AlarmInfo" -and $CloudEvent.Subject -eq "AlarmStatusChangedEvent") {
```

Comment:
They're sent just not processed since this function expects the enriched alarm and hence the comment

Reply:
The alarm server processes ALL alarm events, incl. reset, acknowledged, etc. and injects the details (AlarmInfo).

Reply:
Ahhhh, so I'd probably want to add another AND statement to check for this particular alarm to ensure other Alarms don't magically make it into the processing

Reply:
Exactly. Did you also mention this in the README, i.e. that this function only forwards a specific vSphere alarm event?

Reply:
No, I'll add a quick blurb. I think you meant to say it forwards all vSphere Alarm Events, since there's only a single type. The way the `handler.ps1` will limit it to just enriched BUT you'd still get them all unless you filter further